pkg/resources/grant_helpers.go

```diff
@@ -52,24 +53,67 @@ type grant struct {
 	GrantOption bool
 }
 
-// splitGrantID takes the <db_name>|<schema_name>|<view_name>|<privilege> ID and
-// returns the object name and privilege.
-func splitGrantID(v string) (string, string, string, string, error) {
-	reader := csv.NewReader(strings.NewReader(v))
-	reader.Comma = '|'
+// grantID contains identifying elements that allow unique access privileges
+type grantID struct {
+	ResourceName string
+	SchemaName   string
+	ViewName     string
+	Privilege    string
+}
+
+func (grantID *grantID) String() (string, error) {
+	var buf bytes.Buffer
+	csvWriter := csv.NewWriter(&buf)
+	csvWriter.Comma = '|'
```

Comment:
I would proabably make '|' a constant since it is also used in the read function and they must both be the same

Reply:
Should I make it a global variable of some sort? Not sure how to define & place the constant